### PR TITLE
Feature/namespace import

### PR DESCRIPTION
**Pull request by ChathurangaKCD**

### CODE REVIEW

File: src/commands/apply.ts [hunk 1]
```diff
@@ -185,7 +185,7 @@ async function applyManifests(
         }
         return null;
       }, 1),
-      filter((m): m is ManifestObject => m !== null),
+      filter((m): m is ManifestFileObject => m !== null),
```
Comment (binura-g) [inline src/commands/apply.ts:188]: I'm not familiar with `is` but wouldn't the type guard throw an error here?

Comment (ChathurangaKCD) [reply src/commands/apply.ts:188]: no, `m is ManifestFileObject` narrows down the type to `ManifestFileObject` as long as the returned value is `true`.

Comment (binura-g) [reply src/commands/apply.ts:188]: Oh I see. Gg

File: src/modules/gitops/manifest-import-types.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    description: 'configuration',
+    types: [
+      { kind: 'Secret', apiVersion: 'v1' },
+      { kind: 'ConfigMap', apiVersion: 'v1' },
+    ],
+  },
+  2: {
+    description: 'deployments',
+    types: [
+      { kind: 'Deployment', apiVersion: 'apps/v1' },
```
[the remark this answers is not in the document]
Comment (ChathurangaKCD) [reply src/modules/gitops/manifest-import-types.ts:32]: done

File: src/modules/gitops/manifest-import-types.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  {
+    description: string;
+    types: Array<ResourceType>;
+  }
+>;
+const importTypes: ImportTypes = {
+  1: {
+    description: 'configuration',
+    types: [
+      { kind: 'Secret', apiVersion: 'v1' },
```
[the remark this answers is not in the document]
Comment (ChathurangaKCD) [reply src/modules/gitops/manifest-import-types.ts:22]: done

File: src/modules/gitops/manifest-import-types.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  apiVersion: string;
+}
+export type ImportTypes = Record<
+  ManifestImportPriorities,
+  {
+    description: string;
+    types: Array<ResourceType>;
+  }
+>;
+const importTypes: ImportTypes = {
```
[the remark this answers is not in the document]
Comment (ChathurangaKCD) [reply src/modules/gitops/manifest-import-types.ts]: done

**Review by nilesh93 — APPROVED** [review]

**Pull request merged**